### PR TITLE
Adds Subnav, PageHeader components, implements on add-ons pages.

### PR DESCRIPTION
**Pull request by chuckharmston**
Minor stuff, but I clearly paid deep attention to the aesthetics.

r? @ngokevin 

screen shot 2015-08-28 at 11 11 47 am

### CODE REVIEW

File: src/media/css/pageheader.styl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  align-items flex-end
+  border-bottom 1px solid #DDD
+  display flex
+  justify-content space-between
+  
+  h1
+    font-size 28px
+    font-weight 400
+    margin 0
+
```
Comment (ngokevin) [inline src/media/css/pageheader.styl]: maybe we should start double line-breaking between groups? it's starting to get hard to parse since a single-line break doesn't have as much meaning in terms of logical separation

Comment (chuckharmston) [inline src/media/css/pageheader.styl]: 2 lines for first-level indentation groups, 1 line for everything else? I'm all aboard on that :bullettrain_side:

File: src/media/css/header.styl [hunk 2]
```diff
@@ -1,7 +1,7 @@
 @import 'lib';
 
 
-header
+.app > header
```
Comment (ngokevin) [inline src/media/css/header.styl]: .app-header maybe

File: src/media/js/addon/components/addonSubnav.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import {ReverseLink} from 'react-router-reverse';
+
+import {Subnav} from '../../site/components/subnav';
+
+
+export default class AddonSubnav extends React.Component {
+  render() {
+    return (
+      <Subnav>
+        <ReverseLink to="addon-dashboard">My Add-ons</ReverseLink>
```
Comment (ngokevin) [inline src/media/js/addon/components/addonSubnav.js]: i'd call them FirefoxOS add-ons

File: src/media/js/addon/components/addonSubnav.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        <ReverseLink to="addon-submit">Submit an Add-on</ReverseLink>
```
Comment (ngokevin) [inline src/media/js/addon/components/addonSubnav.js]: current order on the add-on landing is Dashboard/Review/Submit, alphabetical. if we change it here, we should change it there. or just match here to there

Comment (ngokevin) [inline src/media/js/addon/components/addonSubnav.js]: never mind about my previous deleted comment. maybe these should be alphabetical?

File: src/media/js/addon/containers/submit.js [hunk 5]
```diff
@@ -4,6 +4,8 @@ import {bindActionCreators} from 'redux';
 import FileReaderInput from 'react-file-reader-input';
 import {connect} from 'react-redux';
 
+import AddonSubnav from '../components/addonSubnav';
+import PageHeader from '../../site/components/pageHeader';
```
Comment (ngokevin) [inline src/media/js/addon/containers/submit.js]: i'd move this down one line. the way i'm ordering imports is by the latter section. so it would start importing from the current directory, then the parent, and so on, all alphabetically.

File: src/media/js/addon/containers/submit.js [hunk 6]
```diff
@@ -58,8 +60,7 @@ export class AddonSubmit extends React.Component {
   render() {
     return (
       <section>
-        <h1>Submitting a Firefox OS Add-on</h1>
-
```
Comment (ngokevin) [inline src/media/js/addon/containers/submit.js]: might've kept a line-break for logical separation, the form section is pretty big

File: src/media/js/site/components/pageHeader.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import classnames from 'classnames';
+import {connect} from 'react-redux';
+import React from 'react';
+import {reverse, ReverseLink} from 'react-router-reverse';
+
+
+export default class PageHeader extends React.Component {
+  render() {
```
Comment (ngokevin) [inline src/media/js/site/components/pageHeader.js]: proptypes would be good

File: src/media/js/site/components/pageHeader.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import classnames from 'classnames';
+import {connect} from 'react-redux';
```
Comment (ngokevin) [inline src/media/js/site/components/pageHeader.js]: i find it's better to order by the latter section since the former section is hard to order alphabetically due to named imports.

```js
'classnames'
'react'
'react-redux'
'react-router-reverse'
```

File: src/media/js/site/components/subnav.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    return (
+      <li>{navItem}</li>
+    )
+  }
+
+  render() {
+    return (
+      <nav className="subnav">
+        <ul>
+          {this.props.children.map(this.renderNavItem)}
```
Comment (ngokevin) [inline src/media/js/site/components/subnav.js]: if we aren't planning to do much more with the item, maybeee:

```js
{this.props.children.map(item => <li>{item}</li>)}
```

**Comment by ngokevin** [comment]
r+wc, very welcome change! DRY is nice

File: src/media/css/header.styl (hunk 2 above)
Comment (chuckharmston) [inline src/media/css/header.styl]: Just `.header` to follow the `.header--workmark` pattern?

Comment (ngokevin) [inline src/media/css/header.styl]: ok

File: src/media/css/pageheader.styl (hunk 1 above)
Comment (ngokevin) [inline src/media/css/pageheader.styl]: ok
